code/iaas/allocator/src/main/java/io/cattle/platform/allocator/service/AbstractAllocator.java

```diff
@@ -146,8 +149,12 @@ protected void deallocateInstance(final AllocationRequest request) {
     protected void allocateInstance(final AllocationRequest request) {
         final Instance origInstance = objectManager.loadResource(Instance.class, request.getResourceId());
         final List<Instance> instances = getInstancesToAllocate(origInstance);
+        final List<Long> volumeIds = new ArrayList<>();
+        for (Instance instance : instances) {
+            volumeIds.addAll(InstanceHelpers.extractVolumeIdsFromMounts(instance));
```

Comment:
Nit: would it make sense to make volumeIds a set? If the same volumeId can be referenced by diff instances, duplicates are possible